brilift/src/main.rs

```diff
@@ -7,14 +7,62 @@ use cranelift_codegen::settings::Configurable;
 use cranelift_codegen::{ir, isa, settings};
 use cranelift_frontend::{FunctionBuilder, FunctionBuilderContext, Variable};
 use cranelift_jit::{JITBuilder, JITModule};
-use cranelift_module::{default_libcall_names, Module};
+use cranelift_module::{default_libcall_names, FuncOrDataId, Module};
 use cranelift_object::{ObjectBuilder, ObjectModule};
-use enum_map::{enum_map, Enum, EnumMap};
 use std::collections::HashMap;
 use std::fs;
 
-/// Runtime functions used by ordinary Bril instructions.
-#[derive(Debug, Enum)]
+trait RTFuncUtils {
+    fn sig(
+        &self,
+        pointer_type: ir::Type,
+        call_conv: cranelift_codegen::isa::CallConv,
+    ) -> ir::Signature;
+
+    fn name(&self) -> &'static str;
+
+    fn get_func_id<M: Module>(
+        &self,
+        module: &mut M,
+        call_conv: cranelift_codegen::isa::CallConv,
+    ) -> cranelift_module::FuncId {
+        match module.get_name(self.name()) {
```

Comment:
If one wants to continue using Cranelift's philosophy with https://docs.rs/cranelift-codegen/0.84.0/x86_64-apple-darwin/src/cranelift_codegen/ir/libcall.rs.html#141-166 then I guess it would be pretty straight forward to implement the behavior of EnumMap without the dependency. Maybe this would be the way to go given your feedback.

I'm not totally clear on this distinction that is being drawn with using the enum as a source of truth vs a string. Since all of this functionality is enclosed in an `impl` of the enum where the string name is never exposed to/provided by the user, then from an interface perspective, it doesn't look like there is a difference between what I've done and the current structure. I understand the implementation is different, but this is just a consequence of Cranelift under the hood mapping strings to functions. This is also something the current implementation has to do with `func_refs: HashMap<String, ir::FuncRef>`.

I came at this from a different view of efficiency which may in practice be less optimal(We know that only certain functions will be used, why are we paying the cost to declare functions that will never be used and then making additional allocations to store this information?).